Mathlib/Data/List/Lattice.lean

```diff
@@ -159,68 +162,38 @@ end Inter
 
 section BagInter
 
-@[simp]
+@[simp, grind =]
 theorem nil_bagInter (l : List α) : [].bagInter l = [] := by cases l <;> rfl
 
-@[simp]
+@[simp, grind =]
 theorem bagInter_nil (l : List α) : l.bagInter [] = [] := by cases l <;> rfl
 
 @[simp]
 theorem cons_bagInter_of_pos (l₁ : List α) (h : a ∈ l₂) :
     (a :: l₁).bagInter l₂ = a :: l₁.bagInter (l₂.erase a) := by
-  cases l₂
-  · exact if_pos h
-  · simp only [List.bagInter, if_pos (elem_eq_true_of_mem h)]
+  cases l₂ with grind [List.bagInter]
 
 @[simp]
 theorem cons_bagInter_of_neg (l₁ : List α) (h : a ∉ l₂) :
     (a :: l₁).bagInter l₂ = l₁.bagInter l₂ := by
-  cases l₂; · simp only [bagInter_nil]
-  simp only [erase_of_not_mem h, List.bagInter, if_neg (mt mem_of_elem_eq_true h)]
+  cases l₂ with grind [List.bagInter]
+
+@[grind =]
+theorem cons_bagInteger :
+    (a :: l₁).bagInter l₂ = if a ∈ l₂ then a :: l₁.bagInter (l₂.erase a) else l₁.bagInter l₂ := by
```

Comment:
Is the general rule with `grind` that unlike `simp`, we should encourage expansion of `if`s? Could this be document somewhere (perhaps including but not limited to the commit message)?